<h1> Data Visualization Project
<h2> HIV Infection in Children and Adolescent and GDP

<h3> by Ben Benarivo (bb3052@columbia.edu)

In this project, I would like to understand the relationship between HIV infection in children and adolescent (Age group 0 to 19) and GDP. The visualization will be done using Tableau. However, the exploratory data analysis (EDA) and data cleaning will be done using Python on this notebook.

If you want to go directly to the visualization result on Tableau, please click on this <a href="https://public.tableau.com/app/profile/benarivo/viz/GDPvsHIVAidsInfectionsonPopulationAge0-19/Dashboard1">link</a> 

In [100]:
import pandas as pd

<h2> 1.a. HIV Infections for Children and Adolescent EDA

The data I will use for this section is from unicef.org.

In [114]:
df_hiv = pd.read_excel("HIV_Epidemiology.xlsx", "Data", header = 1)

In [124]:
df_hiv

ISO3    Type           Country/Region UNICEF Region  \
0        03M49WLD  Region                   Global           NaN   
1        03M49WLD  Region                   Global           NaN   
2        03M49WLD  Region                   Global           NaN   
3        03M49WLD  Region                   Global           NaN   
4        03M49WLD  Region                   Global           NaN   
...           ...     ...                      ...           ...   
194043  regionWCA  Region  West and Central Africa           NaN   
194044  regionWCA  Region  West and Central Africa           NaN   
194045  regionWCA  Region  West and Central Africa           NaN   
194046  regionWCA  Region  West and Central Africa           NaN   
194047  regionWCA  Region  West and Central Africa           NaN   

                                                Indicator  \
0       Estimated incidence rate (new HIV infection pe...   
1       Estimated incidence rate (new HIV infection pe...   
2       Estimated incidence rate (new HIV infection pe...   
3       Estimated incidence rate (new HIV infection pe...   
4       Estimated incidence rate (new HIV infection pe...   
...                                                   ...   
194043  Estimated rate of annual AIDS-related deaths (...   
194044  Estimated rate of annual AIDS-related deaths (...   
194045  Estimated rate of annual AIDS-related deaths (...   
194046  Estimated rate of annual AIDS-related deaths (...   
194047  Estimated rate of annual AIDS-related deaths (...   

                  Data source  Year     Sex        Age  Value  Lower  Upper  
0       UNAIDS 2022 estimates  2000    Both   Age 0-14   0.86   0.58   1.28  
1       UNAIDS 2022 estimates  2000    Both  Age 10-19   0.70   0.21   1.33  
2       UNAIDS 2022 estimates  2000  Female   Age 0-14   0.87   0.59   1.29  
3       UNAIDS 2022 estimates  2000  Female  Age 10-19   1.04   0.22   1.91  
4       UNAIDS 2022 estimates  2000    Male   Age 0-14   0.85   0.58   1.26  
...                       ...   ...     ...        ...    ...    ...    ...  
194043  UNAIDS 2022 estimates  2021    Both  Age 10-19   6.72   5.35   8.17  
194044  UNAIDS 2022 estimates  2021  Female   Age 0-14  14.49  10.95  18.32  
194045  UNAIDS 2022 estimates  2021  Female  Age 10-19   6.66   5.08   8.25  
194046  UNAIDS 2022 estimates  2021    Male   Age 0-14  14.60  11.05  18.45  
194047  UNAIDS 2022 estimates  2021    Male  Age 10-19   6.77   5.55   8.11  

[194048 rows x 12 columns]

In [125]:
df_hiv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 194048 entries, 0 to 194047
Data columns (total 12 columns):
 #   Column          Non-Null Count   Dtype 
---  ------          --------------   ----- 
 0   ISO3            194048 non-null  object
 1   Type            194048 non-null  object
 2   Country/Region  194048 non-null  object
 3   UNICEF Region   179354 non-null  object
 4   Indicator       194048 non-null  object
 5   Data source     194048 non-null  object
 6   Year            194048 non-null  int64 
 7   Sex             194048 non-null  object
 8   Age             194048 non-null  object
 9   Value           194026 non-null  object
 10  Lower           193770 non-null  object
 11  Upper           193770 non-null  object
dtypes: int64(1), object(11)
memory usage: 17.8+ MB


In [126]:
df_hiv.ISO3.unique()

array(['03M49WLD', 'AFG', 'AGO', 'ALB', 'ARG', 'ARM', 'AUS', 'AZE', 'BDI',
       'BEN', 'BFA', 'BGD', 'BGR', 'BHR', 'BHS', 'BLR', 'BLZ', 'BOL',
       'BRB', 'BTN', 'BWA', 'CAF', 'CHL', 'CIV', 'CMR', 'COD', 'COG',
       'COL', 'COM', 'CPV', 'CRI', 'CUB', 'CYP', 'DEU', 'DJI', 'DNK',
       'DOM', 'DZA', 'ECU', 'EGY', 'ERI', 'ESP', 'ETH', 'FJI', 'FRA',
       'GAB', 'GEO', 'GHA', 'GIN', 'GMB', 'GNB', 'GNQ', 'GRC', 'GTM',
       'GUY', 'HND', 'HRV', 'HTI', 'IDN', 'IND', 'IRL', 'IRN', 'ISL',
       'ITA', 'JAM', 'KAZ', 'KEN', 'KGZ', 'KHM', 'LAO', 'LBN', 'LBR',
       'LBY', 'LKA', 'LSO', 'LTU', 'LUX', 'LVA', 'MAR', 'MDA', 'MDG',
       'MEX', 'MLI', 'MMR', 'MNE', 'MNG', 'MRT', 'MUS', 'MWI', 'MYS',
       'NAM', 'NER', 'NGA', 'NIC', 'NLD', 'NOR', 'NPL', 'NZL', 'OMN',
       'PAK', 'PER', 'PHL', 'PNG', 'PRT', 'PRY', 'QAT', 'ROU', 'RWA',
       'SDN', 'SEN', 'SGP', 'SLE', 'SLV', 'SOM', 'SRB', 'SSD', 'STP',
       'SUR', 'SVK', 'SVN', 'SWZ', 'SYR', 'TCD', 'TGO', 'THA', 'TJK',
       'TLS', '

As we can see that ISO3 is the three-letter country codes representing a country or region. However, there are several data points that are not using the ISO3 standard such as '03M49WLD' for 'The world'

In [127]:
df_hiv["Type"].unique()

array(['Region', 'Country'], dtype=object)

The data is grouped to a regional data and country data

In [128]:
df_hiv["Country/Region"].unique()

array(['Global', 'Afghanistan', 'Angola', 'Albania', 'Argentina',
       'Armenia', 'Australia', 'Azerbaijan', 'Burundi', 'Benin',
       'Burkina Faso', 'Bangladesh', 'Bulgaria', 'Bahrain', 'Bahamas',
       'Belarus', 'Belize', 'Bolivia (Plurinational State of)',
       'Barbados', 'Bhutan', 'Botswana', 'Central African Republic',
       'Chile', "Côte d'Ivoire", 'Cameroon',
       'Democratic Republic of the Congo', 'Congo', 'Colombia', 'Comoros',
       'Cabo Verde', 'Costa Rica', 'Cuba', 'Cyprus', 'Germany',
       'Djibouti', 'Denmark', 'Dominican Republic', 'Algeria', 'Ecuador',
       'Egypt', 'Eritrea', 'Spain', 'Ethiopia', 'Fiji', 'France', 'Gabon',
       'Georgia', 'Ghana', 'Guinea', 'Gambia', 'Guinea-Bissau',
       'Equatorial Guinea', 'Greece', 'Guatemala', 'Guyana', 'Honduras',
       'Croatia', 'Haiti', 'Indonesia', 'India', 'Ireland',
       'Iran (Islamic Republic of)', 'Iceland', 'Italy', 'Jamaica',
       'Kazakhstan', 'Kenya', 'Kyrgyzstan', 'Cambodia',
       "Lao

In [129]:
df_hiv["UNICEF Region"].unique()

array([nan, 'South Asia', 'Eastern and Southern Africa',
       'Eastern Europe and Central Asia', 'Latin America and Caribbean',
       'East Asia and Pacific', 'West and Central Africa',
       'Middle East and North Africa', 'Western Europe'], dtype=object)

In [130]:
df_hiv[df_hiv["Type"] == "Region"]["UNICEF Region"].unique()

array([nan], dtype=object)

All the UNICEF Region for regional data are empty / NULL

In [131]:
df_hiv["Type"].unique()

array(['Region', 'Country'], dtype=object)

In [132]:
df_hiv.Indicator.unique()

array(['Estimated incidence rate (new HIV infection per 1,000 uninfected population)',
       'Estimated mother-to-child transmission rate (%)',
       'Estimated number of annual AIDS-related deaths',
       'Estimated number of annual new HIV infections',
       'Estimated number of people living with HIV',
       'Estimated rate of annual AIDS-related deaths (per 100,000 population)'],
      dtype=object)

These are the different indicators used in the data

In [133]:
df_hiv["Data source"].unique()

array(['UNAIDS 2022 estimates'], dtype=object)

All the data is from UNAIDS 2022 estimates

In [134]:
df_hiv.Year.unique()

array([2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010,
       2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021],
      dtype=int64)

The data is from 200 until 2021

In [135]:
df_hiv.Sex.unique()

array(['Both', 'Female', 'Male'], dtype=object)

The data is also aggregated based on gender. The data for "both" gender are either the summation or the average of the male and female gender data depending on the type of the value/Indicator

In [136]:
df_hiv[(df_hiv["Country/Region"] == "Egypt") & (df_hiv.Age == "Age 0-14") & (df_hiv.Year == 2000)]

ISO3     Type Country/Region                 UNICEF Region  \
51931  EGY  Country          Egypt  Middle East and North Africa   
51939  EGY  Country          Egypt  Middle East and North Africa   
51947  EGY  Country          Egypt  Middle East and North Africa   
52459  EGY  Country          Egypt  Middle East and North Africa   
52467  EGY  Country          Egypt  Middle East and North Africa   
52475  EGY  Country          Egypt  Middle East and North Africa   
52987  EGY  Country          Egypt  Middle East and North Africa   
52989  EGY  Country          Egypt  Middle East and North Africa   
52991  EGY  Country          Egypt  Middle East and North Africa   

                                               Indicator  \
51931     Estimated number of annual AIDS-related deaths   
51939     Estimated number of annual AIDS-related deaths   
51947     Estimated number of annual AIDS-related deaths   
52459         Estimated number of people living with HIV   
52467         Estimated number of people living with HIV   
52475         Estimated number of people living with HIV   
52987  Estimated rate of annual AIDS-related deaths (...   
52989  Estimated rate of annual AIDS-related deaths (...   
52991  Estimated rate of annual AIDS-related deaths (...   

                 Data source  Year     Sex       Age Value Lower Upper  
51931  UNAIDS 2022 estimates  2000    Both  Age 0-14  <100  <100  <100  
51939  UNAIDS 2022 estimates  2000  Female  Age 0-14  <100  <100  <100  
51947  UNAIDS 2022 estimates  2000    Male  Age 0-14  <100  <100  <100  
52459  UNAIDS 2022 estimates  2000    Both  Age 0-14  <200  <200  <200  
52467  UNAIDS 2022 estimates  2000  Female  Age 0-14  <100  <100  <100  
52475  UNAIDS 2022 estimates  2000    Male  Age 0-14  <100  <100  <100  
52987  UNAIDS 2022 estimates  2000    Both  Age 0-14  0.11  0.09  0.14  
52989  UNAIDS 2022 estimates  2000  Female  Age 0-14  0.11  0.09  0.14  
52991  UNAIDS 2022 estimates  2000    Male  Age 0-14  0.12  0.09  0.14

In [137]:
df_hiv[(df_hiv["Country/Region"] == "Indonesia") & (df_hiv.Age == "Age 0-14") & (df_hiv.Year == 2000)]

ISO3     Type Country/Region          UNICEF Region  \
78243  IDN  Country      Indonesia  East Asia and Pacific   
78245  IDN  Country      Indonesia  East Asia and Pacific   
78247  IDN  Country      Indonesia  East Asia and Pacific   
78397  IDN  Country      Indonesia  East Asia and Pacific   
78405  IDN  Country      Indonesia  East Asia and Pacific   
78413  IDN  Country      Indonesia  East Asia and Pacific   
78925  IDN  Country      Indonesia  East Asia and Pacific   
78931  IDN  Country      Indonesia  East Asia and Pacific   
78937  IDN  Country      Indonesia  East Asia and Pacific   
79321  IDN  Country      Indonesia  East Asia and Pacific   
79329  IDN  Country      Indonesia  East Asia and Pacific   
79337  IDN  Country      Indonesia  East Asia and Pacific   
79849  IDN  Country      Indonesia  East Asia and Pacific   
79851  IDN  Country      Indonesia  East Asia and Pacific   
79853  IDN  Country      Indonesia  East Asia and Pacific   

                                               Indicator  \
78243  Estimated incidence rate (new HIV infection pe...   
78245  Estimated incidence rate (new HIV infection pe...   
78247  Estimated incidence rate (new HIV infection pe...   
78397     Estimated number of annual AIDS-related deaths   
78405     Estimated number of annual AIDS-related deaths   
78413     Estimated number of annual AIDS-related deaths   
78925      Estimated number of annual new HIV infections   
78931      Estimated number of annual new HIV infections   
78937      Estimated number of annual new HIV infections   
79321         Estimated number of people living with HIV   
79329         Estimated number of people living with HIV   
79337         Estimated number of people living with HIV   
79849  Estimated rate of annual AIDS-related deaths (...   
79851  Estimated rate of annual AIDS-related deaths (...   
79853  Estimated rate of annual AIDS-related deaths (...   

                 Data source  Year     Sex       Age Value Lower Upper  
78243  UNAIDS 2022 estimates  2000    Both  Age 0-14  0.01  0.01  0.01  
78245  UNAIDS 2022 estimates  2000  Female  Age 0-14  0.01  0.01  0.01  
78247  UNAIDS 2022 estimates  2000    Male  Age 0-14  0.01  0.01  0.01  
78397  UNAIDS 2022 estimates  2000    Both  Age 0-14  <200  <200  <200  
78405  UNAIDS 2022 estimates  2000  Female  Age 0-14  <100  <100  <100  
78413  UNAIDS 2022 estimates  2000    Male  Age 0-14  <100  <100  <100  
78925  UNAIDS 2022 estimates  2000    Both  Age 0-14  <500  <500  <500  
78931  UNAIDS 2022 estimates  2000  Female  Age 0-14  <200  <200  <200  
78937  UNAIDS 2022 estimates  2000    Male  Age 0-14  <200  <200  <200  
79321  UNAIDS 2022 estimates  2000    Both  Age 0-14  <500  <500  <500  
79329  UNAIDS 2022 estimates  2000  Female  Age 0-14  <500  <200  <500  
79337  UNAIDS 2022 estimates  2000    Male  Age 0-14  <500  <200  <500  
79849  UNAIDS 2022 estimates  2000    Both  Age 0-14  0.20  0.17  0.22  
79851  UNAIDS 2022 estimates  2000  Female  Age 0-14  0.19  0.17  0.21  
79853  UNAIDS 2022 estimates  2000    Male  Age 0-14  0.20  0.17  0.22

For the data using absolute values, the 'Value' for both gender is the summation from the 'Value' from male and female data. For the data that use rate, the 'Value' for both gender is the average for the 'Value' from male and female data

In [138]:
df_hiv[(df_hiv["Country/Region"] == "Indonesia") & (df_hiv.Sex == "Both") & (df_hiv.Year == 2000)]

ISO3     Type Country/Region          UNICEF Region  \
78243  IDN  Country      Indonesia  East Asia and Pacific   
78244  IDN  Country      Indonesia  East Asia and Pacific   
78375  IDN  Country      Indonesia  East Asia and Pacific   
78397  IDN  Country      Indonesia  East Asia and Pacific   
78398  IDN  Country      Indonesia  East Asia and Pacific   
78399  IDN  Country      Indonesia  East Asia and Pacific   
78400  IDN  Country      Indonesia  East Asia and Pacific   
78401  IDN  Country      Indonesia  East Asia and Pacific   
78402  IDN  Country      Indonesia  East Asia and Pacific   
78403  IDN  Country      Indonesia  East Asia and Pacific   
78404  IDN  Country      Indonesia  East Asia and Pacific   
78925  IDN  Country      Indonesia  East Asia and Pacific   
78926  IDN  Country      Indonesia  East Asia and Pacific   
78927  IDN  Country      Indonesia  East Asia and Pacific   
78928  IDN  Country      Indonesia  East Asia and Pacific   
78929  IDN  Country      Indonesia  East Asia and Pacific   
78930  IDN  Country      Indonesia  East Asia and Pacific   
79321  IDN  Country      Indonesia  East Asia and Pacific   
79322  IDN  Country      Indonesia  East Asia and Pacific   
79323  IDN  Country      Indonesia  East Asia and Pacific   
79324  IDN  Country      Indonesia  East Asia and Pacific   
79325  IDN  Country      Indonesia  East Asia and Pacific   
79326  IDN  Country      Indonesia  East Asia and Pacific   
79327  IDN  Country      Indonesia  East Asia and Pacific   
79328  IDN  Country      Indonesia  East Asia and Pacific   
79849  IDN  Country      Indonesia  East Asia and Pacific   
79850  IDN  Country      Indonesia  East Asia and Pacific   

                                               Indicator  \
78243  Estimated incidence rate (new HIV infection pe...   
78244  Estimated incidence rate (new HIV infection pe...   
78375    Estimated mother-to-child transmission rate (%)   
78397     Estimated number of annual AIDS-related deaths   
78398     Estimated number of annual AIDS-related deaths   
78399     Estimated number of annual AIDS-related deaths   
78400     Estimated number of annual AIDS-related deaths   
78401     Estimated number of annual AIDS-related deaths   
78402     Estimated number of annual AIDS-related deaths   
78403     Estimated number of annual AIDS-related deaths   
78404     Estimated number of annual AIDS-related deaths   
78925      Estimated number of annual new HIV infections   
78926      Estimated number of annual new HIV infections   
78927      Estimated number of annual new HIV infections   
78928      Estimated number of annual new HIV infections   
78929      Estimated number of annual new HIV infections   
78930      Estimated number of annual new HIV infections   
79321         Estimated number of people living with HIV   
79322         Estimated number of people living with HIV   
79323         Estimated number of people living with HIV   
79324         Estimated number of people living with HIV   
79325         Estimated number of people living with HIV   
79326         Estimated number of people living with HIV   
79327         Estimated number of people living with HIV   
79328         Estimated number of people living with HIV   
79849  Estimated rate of annual AIDS-related deaths (...   
79850  Estimated rate of annual AIDS-related deaths (...   

                 Data source  Year   Sex        Age  Value  Lower  Upper  
78243  UNAIDS 2022 estimates  2000  Both   Age 0-14   0.01   0.01   0.01  
78244  UNAIDS 2022 estimates  2000  Both  Age 10-19   0.09   0.04   0.17  
78375  UNAIDS 2022 estimates  2000  Both    Age 0-4  47.53  44.51  51.23  
78397  UNAIDS 2022 estimates  2000  Both   Age 0-14   <200   <200   <200  
78398  UNAIDS 2022 estimates  2000  Both   Age 0-19   <200   <200   <200  
78399  UNAIDS 2022 estimates  2000  Both    Age 0-4   <200   <200   <200  
78400  UNAIDS 2022 estimates  2000  Both    Age 0-9   <200   <200   <200  
78401  UNAIDS 20

"Lower" and "Upper" are the maximum and minimum range of the "Value".

In [139]:
print(df_hiv["Value"].unique().tolist())

['0.86', '0.70', '0.87', '1.04', '0.85', '0.38', '0.66', '0.98', '0.35', '0.84', '0.62', '0.92', '0.83', '0.33', '0.82', '0.58', '0.81', '0.31', '0.80', '0.54', '0.79', '0.29', '0.77', '0.51', '0.78', '0.76', '0.27', '0.73', '0.49', '0.74', '0.72', '0.26', '0.68', '0.47', '0.69', '0.71', '0.67', '0.25', '0.61', '0.46', '0.24', '0.56', '0.45', '0.57', '0.55', '0.23', '0.43', '0.50', '0.22', '0.44', '0.42', '0.65', '0.21', '0.41', '0.39', '0.63', '0.40', '0.36', '0.19', '0.59', '0.32', '0.18', '0.30', '0.17', '0.16', '0.28', '0.15', '0.48', '0.14', '0.13', '38.04', '37.66', '37.31', '36.92', '36.30', '34.98', '33.49', '31.14', '28.42', '25.91', '23.71', '20.70', '18.18', '16.64', '15.43', '14.59', '14.38', '13.75', '13.03', '12.45', '12.58', '11.94', '340,000', '360,000', '300,000', '330,000', '12,000', '28,000', '15,000', '170,000', '180,000', '150,000', '160,000', '6,200', '16,000', '10,000', '14,000', '6,300', '11,000', '5,100', '350,000', '370,000', '32,000', '17,000', '31,000', '7,6

The values are either numeric or categorical such as <100.

In [140]:
df_hiv["Age"].unique() 

array(['Age 0-14', 'Age 10-19', 'Age 0-4', 'Age 0-19', 'Age 0-9',
       'Age 10-14', 'Age 15-19', 'Age 5-9'], dtype=object)

There are some overlapping in the range of age. We will choose the smallest range which are: Age 0-4, Age 5-9, Age 10-14, Age 15-19 and Age 0-19 to derive overall trends

<h2> 1.b. HIV Infection Data Cleaning 

In this part, I will perform data cleaning to prepare it for visualization in Tableau.

For this project, we only need country-based data. Therefore we will drop the regions and global data.
Additionally, we only need the Age group 0-19

In [141]:
df_hiv = df_hiv[(df_hiv["Type"] == "Country") & (df_hiv["Age"] == "Age 0-19") ]

I also will drop data for both gender because it can be derived by the male and female data

In [142]:
df_hiv = df_hiv[df_hiv["Sex"] != "Both"]

Now we can drop unnecessary columns, which are "Lower","Upper","Data source","UNICEF Region","Type","Age" data

In [143]:
df_hiv = df_hiv.drop(["Lower","Upper","Data source","UNICEF Region","Type","Age"], axis = 1)

There are value with "<", we are going to delete it and assume that the data is about the numerical value. Additionally, we are going to change the type of the "Value" to float.

In [144]:
df_hiv["Value"] = df_hiv["Value"].str.replace("<","")

In [145]:
df_hiv["Value"] = df_hiv["Value"].str.replace(",","")

In [146]:
df_hiv["Value"].astype("float")

1901        100.0
1909        100.0
1925        100.0
1933        100.0
1949        100.0
           ...   
181695    60000.0
181711    64000.0
181719    56000.0
181735    60000.0
181743    53000.0
Name: Value, Length: 13144, dtype: float64

In [147]:
df_hiv_country.Indicator.unique()

array(['Estimated mother-to-child transmission rate (%)',
       'Estimated number of annual AIDS-related deaths',
       'Estimated number of annual new HIV infections',
       'Estimated number of people living with HIV'], dtype=object)

In this project, I want to know the spread only, then I will choose 'Estimated number of annual new HIV infections'.

In [148]:
df_hiv_country_agg = df_hiv[df_hiv.Indicator == 'Estimated number of annual new HIV infections']

In [149]:
df_hiv_country_agg

ISO3 Country/Region                                      Indicator  \
2427    AFG    Afghanistan  Estimated number of annual new HIV infections   
2433    AFG    Afghanistan  Estimated number of annual new HIV infections   
2445    AFG    Afghanistan  Estimated number of annual new HIV infections   
2451    AFG    Afghanistan  Estimated number of annual new HIV infections   
2463    AFG    Afghanistan  Estimated number of annual new HIV infections   
...     ...            ...                                            ...   
181181  ZWE       Zimbabwe  Estimated number of annual new HIV infections   
181193  ZWE       Zimbabwe  Estimated number of annual new HIV infections   
181199  ZWE       Zimbabwe  Estimated number of annual new HIV infections   
181211  ZWE       Zimbabwe  Estimated number of annual new HIV infections   
181217  ZWE       Zimbabwe  Estimated number of annual new HIV infections   

        Year     Sex Value  
2427    2000  Female   100  
2433    2000    Male   100  
2445    2001  Female   100  
2451    2001    Male   100  
2463    2002  Female   100  
...      ...     ...   ...  
181181  2019    Male  2700  
181193  2020  Female  5000  
181199  2020    Male  2800  
181211  2021  Female  4700  
181217  2021    Male  2700  

[4310 rows x 6 columns]

In [151]:
df_hiv_country_agg = df_hiv_country_agg.drop("Indicator", axis = 1)

In [152]:
df_hiv_country_agg

ISO3 Country/Region  Year     Sex Value
2427    AFG    Afghanistan  2000  Female   100
2433    AFG    Afghanistan  2000    Male   100
2445    AFG    Afghanistan  2001  Female   100
2451    AFG    Afghanistan  2001    Male   100
2463    AFG    Afghanistan  2002  Female   100
...     ...            ...   ...     ...   ...
181181  ZWE       Zimbabwe  2019    Male  2700
181193  ZWE       Zimbabwe  2020  Female  5000
181199  ZWE       Zimbabwe  2020    Male  2800
181211  ZWE       Zimbabwe  2021  Female  4700
181217  ZWE       Zimbabwe  2021    Male  2700

[4310 rows x 5 columns]

In [44]:
df_hiv_country_agg.to_csv('df_hiv_country_agg.csv', index=False)

<h2> 2.a. Gross Domestic Production Data EDA

The data used for this project is from the World Bank dataset

In [154]:
df_gdp = pd.read_csv('GDP_Global3.csv')

In [155]:
df_gdp

Series Name     Series Code  \
0                                   GDP (current US$)  NY.GDP.MKTP.CD   
1                                   GDP (current US$)  NY.GDP.MKTP.CD   
2                                   GDP (current US$)  NY.GDP.MKTP.CD   
3                                   GDP (current US$)  NY.GDP.MKTP.CD   
4                                   GDP (current US$)  NY.GDP.MKTP.CD   
..                                                ...             ...   
217                                               NaN             NaN   
218                                               NaN             NaN   
219                                               NaN             NaN   
220  Data from database: World Development Indicators             NaN   
221                          Last Updated: 12/22/2022             NaN   

       Country Name Country Code     2000 [YR2000]     2001 [YR2001]  \
0       Afghanistan          AFG                ..                ..   
1           Albania          ALB  3480355258.04122   3922100793.5403   
2           Algeria          DZA  54790392746.1939  54744712814.8676   
3    American Samoa          ASM                ..                ..   
4           Andorra          AND  1429049198.45218  1546926174.49664   
..              ...          ...               ...               ...   
217             NaN          NaN               NaN               NaN   
218             NaN          NaN               NaN               NaN   
219             NaN          NaN               NaN               NaN   
220             NaN          NaN               NaN               NaN   
221             NaN          NaN               NaN               NaN   

        2002 [YR2002]     2003 [YR2003]     2004 [YR2004]     2005 [YR2005]  \
0    3854235264.37171   4539500884.0479  5220823684.66085  6226199249.32211   
1    4348068242.19512  5611496257.14231  7184685781.51876  8052073539.38058   
2    56760355865.0082  67863828412.6882  85332581188.6107  103198223709.439   
3           512000000         524000000         509000000         500000000   
4    1755910031.99699  2361726862.30248  2894921777.99851  3159905484.39249   
..                ...               ...               ...               ...   
217               NaN               NaN               NaN               NaN   
218               NaN               NaN               NaN               NaN   
219               NaN               NaN               NaN               NaN   
220               NaN               NaN               NaN               NaN   
221               NaN               NaN               NaN               NaN   

     ...     2012 [YR2012]     2013 [YR2013]     2014 [YR2014]  \
0    ...  20203572959.5023  20564485419.1684  20550582746.8448   
1    ...  12319830437.3467  12776220507.0162  13228147516.1168   
2    ...  209058991952.125  209755003250.664  213810024944.464   
3    ...         640000000         638000000         643000000   
4    ...  3188808942.56713  3193704343.20627  3271808157.30038   
..   ...               ...               ...               ...   
217  ...               NaN               NaN               NaN   
218  ...               NaN               NaN               NaN   
219  ...               NaN               NaN               NaN   
220  ...               NaN               NaN               NaN   
221  ...               NaN               NaN               NaN   

        2015 [YR2015]     2016 [YR2016]     2017 [YR2017]     2018 [YR2018]  \
0    19998156214.3988  18019558181.9602    18896352021.94  18418848299.5825   
1    11386850129.8411  11861199830.8396  13019689336.6919  15156432309.8977   
2    165979279263.174  160034163871.455  170097014589.134  174910878623.049   
3           673000000         671000000         612000000         639000000   
4    2789870187.50693  2896679211.86628  3000180750.11297  3218316013.22626   
..                ...               ...               ...               ...   
217  

We can see from the table above that the table holds data of the GDP in USD from 2000 until 2021. However, several last rows are empty or unnecesaries. 

There are several empty GDP data that are filled with ".." too. FOr examples Afghanistan and American Samoa GDP in 2000 and 2001

In [156]:
df_gdp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 222 entries, 0 to 221
Data columns (total 26 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Series Name    219 non-null    object
 1   Series Code    217 non-null    object
 2   Country Name   217 non-null    object
 3   Country Code   217 non-null    object
 4   2000 [YR2000]  217 non-null    object
 5   2001 [YR2001]  217 non-null    object
 6   2002 [YR2002]  217 non-null    object
 7   2003 [YR2003]  217 non-null    object
 8   2004 [YR2004]  217 non-null    object
 9   2005 [YR2005]  217 non-null    object
 10  2006 [YR2006]  217 non-null    object
 11  2007 [YR2007]  217 non-null    object
 12  2008 [YR2008]  217 non-null    object
 13  2009 [YR2009]  217 non-null    object
 14  2010 [YR2010]  217 non-null    object
 15  2011 [YR2011]  217 non-null    object
 16  2012 [YR2012]  217 non-null    object
 17  2013 [YR2013]  217 non-null    object
 18  2014 [YR2014]  217 non-null   

All the data types are object. I will change these in the data cleaning part

<h2> 2.b. Gross Domestic Production Data Data Cleaning

In [160]:
column_names = list(df_gdp.columns.values)

In [161]:
print(column_names)

['Series Name', 'Series Code', 'Country Name', 'Country Code', '2000 [YR2000]', '2001 [YR2001]', '2002 [YR2002]', '2003 [YR2003]', '2004 [YR2004]', '2005 [YR2005]', '2006 [YR2006]', '2007 [YR2007]', '2008 [YR2008]', '2009 [YR2009]', '2010 [YR2010]', '2011 [YR2011]', '2012 [YR2012]', '2013 [YR2013]', '2014 [YR2014]', '2015 [YR2015]', '2016 [YR2016]', '2017 [YR2017]', '2018 [YR2018]', '2019 [YR2019]', '2020 [YR2020]', '2021 [YR2021]']


Let's drop the unnecessary columns

In [162]:
df_gdp

Series Name     Series Code  \
0                                   GDP (current US$)  NY.GDP.MKTP.CD   
1                                   GDP (current US$)  NY.GDP.MKTP.CD   
2                                   GDP (current US$)  NY.GDP.MKTP.CD   
3                                   GDP (current US$)  NY.GDP.MKTP.CD   
4                                   GDP (current US$)  NY.GDP.MKTP.CD   
..                                                ...             ...   
217                                               NaN             NaN   
218                                               NaN             NaN   
219                                               NaN             NaN   
220  Data from database: World Development Indicators             NaN   
221                          Last Updated: 12/22/2022             NaN   

       Country Name Country Code     2000 [YR2000]     2001 [YR2001]  \
0       Afghanistan          AFG                ..                ..   
1           Albania          ALB  3480355258.04122   3922100793.5403   
2           Algeria          DZA  54790392746.1939  54744712814.8676   
3    American Samoa          ASM                ..                ..   
4           Andorra          AND  1429049198.45218  1546926174.49664   
..              ...          ...               ...               ...   
217             NaN          NaN               NaN               NaN   
218             NaN          NaN               NaN               NaN   
219             NaN          NaN               NaN               NaN   
220             NaN          NaN               NaN               NaN   
221             NaN          NaN               NaN               NaN   

        2002 [YR2002]     2003 [YR2003]     2004 [YR2004]     2005 [YR2005]  \
0    3854235264.37171   4539500884.0479  5220823684.66085  6226199249.32211   
1    4348068242.19512  5611496257.14231  7184685781.51876  8052073539.38058   
2    56760355865.0082  67863828412.6882  85332581188.6107  103198223709.439   
3           512000000         524000000         509000000         500000000   
4    1755910031.99699  2361726862.30248  2894921777.99851  3159905484.39249   
..                ...               ...               ...               ...   
217               NaN               NaN               NaN               NaN   
218               NaN               NaN               NaN               NaN   
219               NaN               NaN               NaN               NaN   
220               NaN               NaN               NaN               NaN   
221               NaN               NaN               NaN               NaN   

     ...     2012 [YR2012]     2013 [YR2013]     2014 [YR2014]  \
0    ...  20203572959.5023  20564485419.1684  20550582746.8448   
1    ...  12319830437.3467  12776220507.0162  13228147516.1168   
2    ...  209058991952.125  209755003250.664  213810024944.464   
3    ...         640000000         638000000         643000000   
4    ...  3188808942.56713  3193704343.20627  3271808157.30038   
..   ...               ...               ...               ...   
217  ...               NaN               NaN               NaN   
218  ...               NaN               NaN               NaN   
219  ...               NaN               NaN               NaN   
220  ...               NaN               NaN               NaN   
221  ...               NaN               NaN               NaN   

        2015 [YR2015]     2016 [YR2016]     2017 [YR2017]     2018 [YR2018]  \
0    19998156214.3988  18019558181.9602    18896352021.94  18418848299.5825   
1    11386850129.8411  11861199830.8396  13019689336.6919  15156432309.8977   
2    165979279263.174  160034163871.455  170097014589.134  174910878623.049   
3           673000000         671000000         612000000         639000000   
4    2789870187.50693  2896679211.86628  3000180750.11297  3218316013.22626   
..                ...               ...               ...               ...   
217  

In [163]:
df_gdp = df_gdp.drop(['Series Name', 'Series Code'], axis = 1)

Now, I will drop the unneccary rows, which are row 217 until 221

In [164]:
df_gdp = df_gdp.drop(df_gdp.index[217:])

I want to make the structure of the GDP data to be similar with the HIV infection data, in which I will have a column that has the Year data, instead having the years as columns.

In [165]:
df_gdp = df_gdp.set_index(['Country Name','Country Code'])

In [166]:
df_gdp

2000 [YR2000]     2001 [YR2001]  \
Country Name          Country Code                                       
Afghanistan           AFG                         ..                ..   
Albania               ALB           3480355258.04122   3922100793.5403   
Algeria               DZA           54790392746.1939  54744712814.8676   
American Samoa        ASM                         ..                ..   
Andorra               AND           1429049198.45218  1546926174.49664   
...                                              ...               ...   
Virgin Islands (U.S.) VIR                         ..                ..   
West Bank and Gaza    PSE                 4313600000        4003700000   
Yemen, Rep.           YEM           9652436179.64605  9861560094.74009   
Zambia                ZMB           3600683039.73254  4094480988.11931   
Zimbabwe              ZWE                 6689957600        6777384700   

                                       2002 [YR2002]     2003 [YR2003]  \
Country Name          Country Code                                       
Afghanistan           AFG           3854235264.37171   4539500884.0479   
Albania               ALB           4348068242.19512  5611496257.14231   
Algeria               DZA           56760355865.0082  67863828412.6882   
American Samoa        ASM                  512000000         524000000   
Andorra               AND           1755910031.99699  2361726862.30248   
...                                              ...               ...   
Virgin Islands (U.S.) VIR                 3262000000        3443000000   
West Bank and Gaza    PSE                 3555800000        3968000000   
Yemen, Rep.           YEM           10694628091.6726  11777966673.3897   
Zambia                ZMB           4193845678.17033  4901839731.26571   
Zimbabwe              ZWE                 6342116400        5727591800   

                                       2004 [YR2004]     2005 [YR2005]  \
Country Name          Country Code                                       
Afghanistan           AFG           5220823684.66085  6226199249.32211   
Albania               ALB           7184685781.51876  8052073539.38058   
Algeria               DZA           85332581188.6107  103198223709.439   
American Samoa        ASM                  509000000         500000000   
Andorra               AND           2894921777.99851  3159905484.39249   
...                                              ...               ...   
Virgin Islands (U.S.) VIR                 3797000000        4428000000   
West Bank and Gaza    PSE                 4603100000        5125700000   
Yemen, Rep.           YEM           13872791658.5487  16746344766.2044   
Zambia                ZMB           6221077674.77871  8331870169.14977   
Zimbabwe              ZWE                 5805598400        5755215200   

                                       2006 [YR2006]     2007 [YR2007]  \
Country Name          Country Code                                       
Afghanistan           AFG           6971379033.26327  9715761649.79443   
Albania               ALB            8896072918.9814  10677324144.4289   
Algeria               DZA            117027307540.89   134977082623.78   
American Samoa        ASM                  493000000         518000000   
Andorra               AND             3456442102.622  3952600602.24473   
...                                              ...               ...   
Virgin Islands (U.S.) VIR                 4484000000        4784000000   
West Bank and Gaza    PSE                 5348300000        5815700000   
Yemen, Rep.           YEM           19061978586.1278  21650532264.2322   
Zambia                ZMB           12756858899.2812  14056957976.2648   
Zimbabwe              ZWE                 5443896500        5291950100   

                                       2008 [YR2008]     2009 [YR2009]  ...  \
Country Name          Country Code                                      ...   
Afghanistan           AFG    

In [167]:
df1 = df_gdp.stack()

In [168]:
df1 = df1.to_frame() #Because .stack() output is a Series

In [169]:
df1

0
Country Name Country Code                                
Afghanistan  AFG          2000 [YR2000]                ..
                          2001 [YR2001]                ..
                          2002 [YR2002]  3854235264.37171
                          2003 [YR2003]   4539500884.0479
                          2004 [YR2004]  5220823684.66085
...                                                   ...
Zimbabwe     ZWE          2017 [YR2017]   17584890937.482
                          2018 [YR2018]  34156069918.0609
                          2019 [YR2019]  21832234925.5021
                          2020 [YR2020]  21509698406.1116
                          2021 [YR2021]  28371238665.5116

[4774 rows x 1 columns]

Because the missing GDP data are filled with "..", I will change it to 0 instead of NULL because we do not want to have NULL for our 

In [170]:
df1[df1[0] == ".."] = None

In [172]:
df1 = df1.fillna(0)

In [173]:
df1

0
Country Name Country Code                                
Afghanistan  AFG          2000 [YR2000]                 0
                          2001 [YR2001]                 0
                          2002 [YR2002]  3854235264.37171
                          2003 [YR2003]   4539500884.0479
                          2004 [YR2004]  5220823684.66085
...                                                   ...
Zimbabwe     ZWE          2017 [YR2017]   17584890937.482
                          2018 [YR2018]  34156069918.0609
                          2019 [YR2019]  21832234925.5021
                          2020 [YR2020]  21509698406.1116
                          2021 [YR2021]  28371238665.5116

[4774 rows x 1 columns]

In [174]:
df1 = df1.astype(float)

In [175]:
df1 

0
Country Name Country Code                            
Afghanistan  AFG          2000 [YR2000]  0.000000e+00
                          2001 [YR2001]  0.000000e+00
                          2002 [YR2002]  3.854235e+09
                          2003 [YR2003]  4.539501e+09
                          2004 [YR2004]  5.220824e+09
...                                               ...
Zimbabwe     ZWE          2017 [YR2017]  1.758489e+10
                          2018 [YR2018]  3.415607e+10
                          2019 [YR2019]  2.183223e+10
                          2020 [YR2020]  2.150970e+10
                          2021 [YR2021]  2.837124e+10

[4774 rows x 1 columns]

In [176]:
df1 = df1.reset_index()

In [177]:
df1

Country Name Country Code        level_2             0
0     Afghanistan          AFG  2000 [YR2000]  0.000000e+00
1     Afghanistan          AFG  2001 [YR2001]  0.000000e+00
2     Afghanistan          AFG  2002 [YR2002]  3.854235e+09
3     Afghanistan          AFG  2003 [YR2003]  4.539501e+09
4     Afghanistan          AFG  2004 [YR2004]  5.220824e+09
...           ...          ...            ...           ...
4769     Zimbabwe          ZWE  2017 [YR2017]  1.758489e+10
4770     Zimbabwe          ZWE  2018 [YR2018]  3.415607e+10
4771     Zimbabwe          ZWE  2019 [YR2019]  2.183223e+10
4772     Zimbabwe          ZWE  2020 [YR2020]  2.150970e+10
4773     Zimbabwe          ZWE  2021 [YR2021]  2.837124e+10

[4774 rows x 4 columns]

In [178]:
df1.rename(columns={"level_2": "Year"}, inplace=True)

In [179]:
df1.rename(columns={0: "GDP in bill USD"}, inplace=True)

In [180]:
df1["GDP in bill USD"] = df1["GDP in bill USD"]/1000000000

In [181]:
df1

Country Name Country Code           Year  GDP in bill USD
0     Afghanistan          AFG  2000 [YR2000]         0.000000
1     Afghanistan          AFG  2001 [YR2001]         0.000000
2     Afghanistan          AFG  2002 [YR2002]         3.854235
3     Afghanistan          AFG  2003 [YR2003]         4.539501
4     Afghanistan          AFG  2004 [YR2004]         5.220824
...           ...          ...            ...              ...
4769     Zimbabwe          ZWE  2017 [YR2017]        17.584891
4770     Zimbabwe          ZWE  2018 [YR2018]        34.156070
4771     Zimbabwe          ZWE  2019 [YR2019]        21.832235
4772     Zimbabwe          ZWE  2020 [YR2020]        21.509698
4773     Zimbabwe          ZWE  2021 [YR2021]        28.371239

[4774 rows x 4 columns]

Now, I want to change the data in the "Year" column into integer

In [183]:
df1["Year"] = df1["Year"].str[:4]

In [184]:
df1["Year"] = df1["Year"].astype(int)

In [185]:
df1

Country Name Country Code  Year  GDP in bill USD
0     Afghanistan          AFG  2000         0.000000
1     Afghanistan          AFG  2001         0.000000
2     Afghanistan          AFG  2002         3.854235
3     Afghanistan          AFG  2003         4.539501
4     Afghanistan          AFG  2004         5.220824
...           ...          ...   ...              ...
4769     Zimbabwe          ZWE  2017        17.584891
4770     Zimbabwe          ZWE  2018        34.156070
4771     Zimbabwe          ZWE  2019        21.832235
4772     Zimbabwe          ZWE  2020        21.509698
4773     Zimbabwe          ZWE  2021        28.371239

[4774 rows x 4 columns]

In [186]:
df_gdp = df1

In [187]:
df_gdp

Country Name Country Code  Year  GDP in bill USD
0     Afghanistan          AFG  2000         0.000000
1     Afghanistan          AFG  2001         0.000000
2     Afghanistan          AFG  2002         3.854235
3     Afghanistan          AFG  2003         4.539501
4     Afghanistan          AFG  2004         5.220824
...           ...          ...   ...              ...
4769     Zimbabwe          ZWE  2017        17.584891
4770     Zimbabwe          ZWE  2018        34.156070
4771     Zimbabwe          ZWE  2019        21.832235
4772     Zimbabwe          ZWE  2020        21.509698
4773     Zimbabwe          ZWE  2021        28.371239

[4774 rows x 4 columns]

In [188]:
df_gdp.to_csv('df_gdp.csv', index=False)

I will now create a visualization on Tableau: 

<a href="https://public.tableau.com/app/profile/benarivo/viz/GDPvsHIVAidsInfectionsonPopulationAge0-19/Dashboard1">Tableau Viz</a>

<h2> Thank you!